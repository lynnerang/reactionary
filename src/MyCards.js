import React, { Component } from 'react';
import './MyCards.scss';
import CardForm from './CardForm.js';
import MyCardList from './MyCardList.js';

class MyCards extends Component {
  constructor(props) {
    super(props);

    this.state = {
      showForm: false
    }
  }

  toggleForm = () => {
    const update = !this.state.showForm;
    this.setState({showForm: update})
  }

  cancelForm = () => {
    document.querySelectorAll('.new-card-input').forEach(i => i.value = '');
    this.setState({showForm: false});
  }

  render() {
    const form = this.state.showForm 
    ? <CardForm cancelForm={this.cancelForm} saveNewCard={this.props.saveNewCard} /> 
    : null;

    return (
      <>
        <h2>My Cards</h2>
        <div className='instructions'>
          <p className='top-line'>Add and delete your own custom flash cards below!</p>
          <p className='notes'>NOTE: These will be included in the 'Guess Mode' and 'Review Mode' games only.</p>
        </div>
        <div className='new-card-link'>
          <button type='button' id='new-card-btn' onClick={this.toggleForm}>+ Add New Card</button>
        </div>
        <div className='my-card-list'>
          {form}
          <MyCardList myCards={this.props.myCards} 
                      deleteMyCard={this.props.deleteMyCard} 
          />
        </div>
      </>
    )
  }
}

export default MyCards;










