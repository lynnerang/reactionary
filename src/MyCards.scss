@import 'variables.scss';

.new-card-link {
  display: flex;
  align-items: center;
  margin: 20px 0px;
  color: $secondary-color;
  font-weight: 400;

  &:hover {
    color: $blue;
  }
}

#new-card-btn {
  border: none;
  border-radius: 100%;
  font-size: 22px;
  padding-bottom: 5px;
  color: $secondary-color;

  &:hover {
    color: $blue;
  }
}

.my-card-list {
  display: flex;
  flex-wrap: wrap;

  form,
  .card {
    margin: 0px 20px 40px 20px;
  }
}

.top-line {
  margin-bottom: 10px;
}

.notes {
  font-style: italic;
  color: rgba(60, 60, 60, 0.66);
  font-size: 17px;
}

