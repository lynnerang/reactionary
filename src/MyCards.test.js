import React from 'react';
import MyCards from './MyCards.js';
import { shallow } from 'enzyme';

const mockData = [
  {
  id: 1,
  term: "Single-page Application",
  fakeAnswers: [
  "Webpack",
  "Responsive Website"
  ],
  definition: "An application that loads a single HTML page and all the necessary assets (such as JavaScript and CSS) required for the app to run. Any interactions don't require a trip to server/page reload.",
  example: "./images/example{exampleId}.png"
  },
  {
  id: 2,
  term: "Compilers",
  fakeAnswers: [
  "Bundlers",
  "Package Managers"
  ],
  definition: "These things take JavaScript code, transform it, and return it as JavaScript code in a different format.",
  example: "Babel is a popular example, and is most commonly used for React and transforms ES6 syntax into syntax that older browsers can interpret."
  }]

  const props = {
    saveNewCard: jest.fn(),
    deleteMyCard: jest.fn()
  }

describe('MyCards', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow (
      <MyCards data={mockData}
      myCards={mockData}
      />
      )
  });

  it('Should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('has one default state property for showForm', () => {
    expect(wrapper.state()).toEqual({showForm: false});
  });

  it('can toggle the showForm state property value', () => {
    expect(wrapper.state('showForm')).toEqual(false);

    wrapper.instance().toggleForm();

    expect(wrapper.state('showForm')).toEqual(true);
  });

  it('can cancel a form in progress and turn showForm to false.', () => {
    expect(wrapper.state('showForm')).toEqual(false);

    wrapper.instance().toggleForm();

    expect(wrapper.state('showForm')).toEqual(true);

    wrapper.instance().cancelForm();

    expect(wrapper.state('showForm')).toEqual(false);
  });

});