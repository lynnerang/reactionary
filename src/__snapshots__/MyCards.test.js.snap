// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyCards Should match the snapshot 1`] = `
<Fragment>
  <h2>
    My Cards
  </h2>
  <div
    className="instructions"
  >
    <p
      className="top-line"
    >
      Add and delete your own custom flash cards below!
    </p>
    <p
      className="notes"
    >
      NOTE: These will be included in the 'Guess Mode' and 'Review Mode' games only.
    </p>
  </div>
  <div
    className="new-card-link"
  >
    <button
      id="new-card-btn"
      onClick={[Function]}
      type="button"
    >
      + Add New Card
    </button>
  </div>
  <div
    className="my-card-list"
  >
    <MyCardList
      canDelete={true}
      data={
        Array [
          Object {
            "definition": "An application that loads a single HTML page and all the necessary assets (such as JavaScript and CSS) required for the app to run. Any interactions don't require a trip to server/page reload.",
            "example": "./images/example{exampleId}.png",
            "fakeAnswers": Array [
              "Webpack",
              "Responsive Website",
            ],
            "id": 1,
            "term": "Single-page Application",
          },
          Object {
            "definition": "These things take JavaScript code, transform it, and return it as JavaScript code in a different format.",
            "example": "Babel is a popular example, and is most commonly used for React and transforms ES6 syntax into syntax that older browsers can interpret.",
            "fakeAnswers": Array [
              "Bundlers",
              "Package Managers",
            ],
            "id": 2,
            "term": "Compilers",
          },
        ]
      }
      myCards={
        Array [
          Object {
            "definition": "An application that loads a single HTML page and all the necessary assets (such as JavaScript and CSS) required for the app to run. Any interactions don't require a trip to server/page reload.",
            "example": "./images/example{exampleId}.png",
            "fakeAnswers": Array [
              "Webpack",
              "Responsive Website",
            ],
            "id": 1,
            "term": "Single-page Application",
          },
          Object {
            "definition": "These things take JavaScript code, transform it, and return it as JavaScript code in a different format.",
            "example": "Babel is a popular example, and is most commonly used for React and transforms ES6 syntax into syntax that older browsers can interpret.",
            "fakeAnswers": Array [
              "Bundlers",
              "Package Managers",
            ],
            "id": 2,
            "term": "Compilers",
          },
        ]
      }
    />
  </div>
</Fragment>
`;
